roles/jd-server/src/lib/mempool/mod.rs

```diff
@@ -28,42 +32,58 @@ pub struct TransacrtionWithHash {
 #[derive(Clone, Debug)]
 pub struct JDsMempool {
     pub mempool: Vec<TransacrtionWithHash>,
-    auth: Auth,
+    auth: mini_rpc_client::Auth,
     url: String,
+    receiver: Receiver<String>,
 }
 
 impl JDsMempool {
-    pub fn get_client(&self) -> Option<RpcClient> {
+    pub fn get_client(&self) -> Option<mini_rpc_client::MiniRpcClient> {
         let url = self.url.as_str();
         if url.contains("http") {
-            Some(RpcClient::new(url, self.auth.clone()).unwrap())
+            let client = mini_rpc_client::MiniRpcClient::new(url.to_string(), self.auth.clone());
+            Some(client)
         } else {
             None
         }
     }
-    pub fn new(url: String, username: String, password: String) -> Self {
-        let auth = Auth::UserPass(username, password);
+
+    // This function is used only for debug purposes
```

Comment:
```suggestion
/// This function is used only for debug purposes and should not be used
/// in production code.
```

Reply:
we have the comment and the _ in front so it should be sufficient but if possible add also a conditional compilation flag

Reply:
like keet it out of the binary if in release mode